pylink/library.py

```diff
@@ -84,7 +85,30 @@ class Library(object):
 
     JLINK_SDK_NAME = 'libjlinkarm'
 
-    WINDOWS_JLINK_SDK_NAME = 'JLinkARM'
+    WINDOWS_32_JLINK_SDK_NAME = 'JLinkARM'
+    WINDOWS_64_JLINK_SDK_NAME = 'JLink_x64'
+
+    @classmethod
+    def get_appropriate_windows_sdk_name(cls):
```

Comment:
I wouldn't create a new function to do this.  Just have the call in `find_library_windows()` have an if-statement.  Also, `sys.maxsize` is the correct one I think, so that always returns the interpreter's size.
```
if sys.maxsize > 2 ** 32:
    dll = WINDOWS_64_JLINK_SDK_NAME + '.dll'
else:
    dll = WINDOWS_32_JLINK_SDK_NAME + '.dll'
```

Reply:
I'm surprised somebody refuse a function for duplicated logic - we need that in at least two places in the code.

Reply:
Ah, I see now.  I think it would be better to just get rid of the `_sdk` attribute, and have the `find_library_{platform}` methods call the `ctypes_util.find_library` method themselves on the SDK name, then fallback to the path searching.  That is probably the correct way to begin with.

Reply:
@hkpeprah I don't think I understand what do you want to have done. And I'm deeply sorry, but I don't have more resources to work on refactoring of this PR. I made it working for our QA engineers, but I'm currently very busy on something completely unrelated.

You current code doesn't support 64 bit Python, so it's going to affect most Windows 10 users - probably not very many. Most people probably will not understand the error on attempt, so they'll just surrender the use. I think it's better to have this change as 64 bit support than don't and wait 10 years until I have time to return to this.

Reply:
My system platform is windows 10, x64. I have tested the pylink/library.py, it did not work. The error massage is bekow.
Traceback (most recent call last):
  File "jlink_flash.py", line 40, in <module>
    jlink = pylink.JLink()
  File "D:\APP\Python27\lib\site-packages\pylink\jlink.py", line 249, in __init__
    lib = library.Library()
  File "D:\APP\Python27\lib\site-packages\pylink\library.py", line 266, in __init__
    self.load_default()
  File "D:\APP\Python27\lib\site-packages\pylink\library.py", line 304, in load_default
    return self.load(path)
  File "D:\APP\Python27\lib\site-packages\pylink\library.py", line 358, in load
    self._lib = ctypes.cdll.LoadLibrary(tf.name)
  File "D:\APP\Python27\lib\ctypes\__init__.py", line 440, in LoadLibrary
    return self._dlltype(name)
  File "D:\APP\Python27\lib\ctypes\__init__.py", line 362, in __init__
    self._handle = _dlopen(self._name, mode)
WindowsError: [Error 193] %1 不是有效的 Win32 --->[Error 193] %1 is not valid Win32.

And the output of "sys.maxsize" if 0x7fffffff, is not "2**63 - 1". I will test 'test_library.py' next.

Reply:
The error massage is caused by my python version with win32. Sorry for that.
Python 2.7.13 (v2.7.13:a06454b1afa1, Dec 17 2016, 20:42:59) [MSC v.1500 32 bit (Intel)] on win32